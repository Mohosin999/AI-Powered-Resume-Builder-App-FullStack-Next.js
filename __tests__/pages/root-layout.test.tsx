// __tests__/app/root-layout.test.tsx
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import RootLayout from "@/app/layout";

// Mock ClerkProvider
jest.mock("@clerk/nextjs", () => ({
  ClerkProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="clerk-provider">{children}</div>
  ),
}));

// Mock ThemeProvider
jest.mock("../../src/components/theme-provider", () => ({
  ThemeProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="theme-provider">{children}</div>
  ),
}));

// Mock Navbar
jest.mock("../../src/components/navbar", () => {
  const MockNavbar = () => <div data-testid="mock-navbar" />;
  MockNavbar.displayName = "Navbar";
  return MockNavbar;
});

// Mock Footer
jest.mock("../../src/components/footer", () => {
  const MockFooter = () => <div data-testid="mock-footer" />;
  MockFooter.displayName = "Footer";
  return MockFooter;
});

describe("Root Layout", () => {
  it("renders child inside root layout with navbar and footer", () => {
    render(
      <RootLayout>
        <div data-testid="child">Child</div>
      </RootLayout>
    );

    expect(screen.getByTestId("child")).toBeInTheDocument();
    expect(screen.getByTestId("clerk-provider")).toBeInTheDocument();
    expect(screen.getByTestId("theme-provider")).toBeInTheDocument();
    expect(screen.getByTestId("mock-navbar")).toBeInTheDocument();
    expect(screen.getByTestId("mock-footer")).toBeInTheDocument();
  });
});
