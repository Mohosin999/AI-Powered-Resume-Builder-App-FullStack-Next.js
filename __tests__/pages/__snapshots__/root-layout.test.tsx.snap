// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Root Layout renders child inside root layout with navbar and footer 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="className antialiased bg-[#F4F2EE] dark:bg-[#131A25]"
    >
      <div
        data-testid="clerk-provider"
      >
        <div
          data-testid="theme-provider"
        >
          <div
            id="header-footer"
          >
            <div
              data-testid="mock-navbar"
            />
          </div>
          <div
            class="min-h-screen"
          >
            <div
              data-testid="child"
            >
              Child
            </div>
          </div>
          <div
            id="header-footer"
          >
            <div
              data-testid="mock-footer"
            />
          </div>
          <section
            aria-atomic="false"
            aria-label="Notifications Alt+T"
            aria-live="polite"
            aria-relevant="additions text"
            class="Toastify"
          />
        </div>
      </div>
    </body>
  </html>
</div>
`;
